### PR TITLE
Update Windows and Mac build instructions and dependency versions

### PR DESCRIPTION
**Pull request by micahflee**
I've updated the dependencies (moved to python 3.6.4, newer version of PyInstallers, and other deps too) and the build instructions for Windows (and Mac). I also recreated the NSIS installer script because all the filenames had changed.

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -102,12 +104,12 @@ Download and install the standalone [Windows 10 SDK](https://dev.windows.com/en-
...
+* Go to http://nsis.sourceforge.net/Download and download the latest NSIS. I downloaded `nsis-3.02.1-setup.exe`.
 * Add `C:\Program Files (x86)\NSIS` to the path.
 
 If you want to sign binaries with Authenticode:
 
-* You'll need a code signing certificate. I roughly followed [this guide](http://blog.assarbad.net/20110513/startssl-code-signing-certificate/) to make one using my StartSSL account.
+* You'll need a code signing certificate. I got an open source code signing certificate from [Certum](https://www.certum.eu/certum/cert,offer_en_open_source_cs.xml).
 * Once you get a code signing key and certificate and covert it to a pfx file, import it into your certificate store.
 
 ### To make a .exe:
```
Comment (mig5) [inline BUILD.md]: "in your Python 3.5 folder" should be "in your Python 3.6 folder"

File: BUILD.md [hunk 2]
```diff
@@ -71,15 +71,17 @@ Now you should have `dist/OnionShare.pkg`.
 
 ### Setting up your dev environment
 
-Download the latest Python 3.5.2, 32-bit (x86) from https://www.python.org/downloads/release/python-352/ (note that there's a pyinstaller/pywin32 bug that prevents 3.6.x from working). I downloaded `python-3.5.2.exe`. When installing it, make sure to check the "Add Python 3.5 to PATH" checkbox on the first page of the installer.
+Download Python 3.6.4, 32-bit (x86) from https://www.python.org/downloads/release/python-364/. I downloaded `python-3.6.4.exe`. When installing it, make sure to check the "Add Python 3.5 to PATH" checkbox on the first page of the installer.
```
Comment (mig5) [inline BUILD.md]: "Add Python 3.5 to PATH" should be "Add Python 3.6 to PATH"

**Comment by mig5** [comment]
Sorry, for all sorts of reasons (mainly Windows install hell... not enough disk space to make a virtual machine that will fit all of Qt and everything else) I haven't had a chance to test this yet.

I spotted two references to Python 3.5 in the BUILD.md that probably should be 3.6 now.

I'll try and install and test tomorrow.

File: BUILD.md [hunk 3]
```diff
@@ -102,12 +104,12 @@ Download and install the standalone [Windows 10 SDK](https://dev.windows.com/en-
```
Comment (mig5) [inline BUILD.md]: Python35-32 on line 103 should be Python36-32

**Comment by micahflee** [comment]
Good catch on the Python 3.6 references.

Also I just went through installing Qt process myself yesterday and omg it's ridiculous. I should have warned you, but you can make it much faster if you selectively check on the Qt and x86 components, and ignore everything for all other architectures, for Android, the sources, etc. It ends up using only like 18gb instead of 40gb.

File: BUILD.md [hunk 4]
```diff
@@ -98,31 +100,29 @@ Download and install the standalone [Windows 10 SDK](https://dev.windows.com/en-
...
 
-* You'll need a code signing certificate. I roughly followed [this guide](http://blog.assarbad.net/20110513/startssl-code-signing-certificate/) to make one using my StartSSL account.
+* You'll need a code signing certificate. I got an open source code signing certificate from [Certum](https://www.certum.eu/certum/cert,offer_en_open_source_cs.xml).
 * Once you get a code signing key and certificate and covert it to a pfx file, import it into your certificate store.
 
 ### To make a .exe:
 
-For PyInstaller to work, you might need to edit `Scripts\pyinstaller-script.py` in your Python 3.5 folder, to work around [this bug](https://stackoverflow.com/questions/31808180/installing-pyinstaller-via-pip-leads-to-failed-to-create-process) in pip.
-
 * Open a command prompt, cd into the onionshare directory, and type: `pyinstaller install\pyinstaller.spec`. `onionshare-gui.exe` and all of their supporting files will get created inside the `dist` folder.
```
Comment (mig5) [inline BUILD.md]: I didn't need the trailing dot at the end of 'install\pyinstaller.spec.' - not sure if it's a big deal?

**Comment by mig5** [comment]
@micahflee especially on Australian internet, omg :)

Just tested building the installer exe (and then installing it) and all went well!

I can't seem to get past 80% when connecting with bundled mode, in my Windows install, but I'm not sure that's any issue with this PR, maybe just slow because I'm in a VM or because of Tor congestion. But what's odd is that it sits at 80% forever, whereas I would expect it to timeout after 90s or whatever we've got it set to these days..

Actually, using Tor Browser, it too hangs at around 80% forever so maybe it's just me...

In addition, when I 'cancel' the connection and SettingsDialog opens, and I click Save, I expect the Tor connection to 'reboot' and see the progress bar again, like I would in Linux or Mac, but I don't. That might be a separate bug, do you reproduce?

If you are able to actually perform a share within your Windows, then maybe it's just me and you can ignore this.

**Comment by micahflee** [comment]
@Baccount do you mind following along with the instructions here for Mac, including using python 3.6.4 and Qt 5.10.0, and letting me know if this works for you, and also if it resolves #451 and #533?

And @mig5, do you also mind finishing reviewing this PR for Windows?

File: BUILD.md (hunk 4 above)
Comment (micahflee) [reply BUILD.md]: Oh the dot is just a period at the end of the sentence, it isn't part of the code block :)

Comment (mig5) [reply BUILD.md]: @micahflee ah my bad :) I thought onionshare-gui.exe was a sort of 'output' argument for the pyinstaller command.

**Comment by Baccount** [comment]
@micahflee 
After installing everything, I still get a traceback when the app runs.
```
Traceback (most recent call last):
  File "onionshare-pyinstaller", line 34, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "onionshare_gui/__init__.py", line 22, in <module>
    from .alert import Alert
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "onionshare_gui/alert.py", line 22, in <module>
    from onionshare import common
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "onionshare/__init__.py", line 23, in <module>
    from . import strings, common, web
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "onionshare/web.py", line 32, in <module>
    from flask import (
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "site-packages/flask/__init__.py", line 19, in <module>
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "site-packages/jinja2/__init__.py", line 82, in <module>
  File "site-packages/jinja2/__init__.py", line 78, in _patch_async
ModuleNotFoundError: No module named 'jinja2.asyncsupport'
[180] Failed to execute script onionshare-pyinstaller

```

**Pull request merged**